File: serverless/serverless-python/src/services/dynamodb/currencies.py
import boto3
from ...config import dynamodb_config
from ...utils.object_helpers import transform_currency_item
dynamodb = boto3.resource('dynamodb')


def get_currencies():
  table = dynamodb.Table(dynamodb_config['tableName'])
  response = table.scan()
  data = response['Items']
  while 'LastEvaluatedKey' in response:
    response = table.scan(ExclusiveStartKey=response['LastEvaluatedKey'])
    data.extend(response['Items'])
  return response['Items']


def put_currency(params):
  item = transform_currency_item(params)
  table = dynamodb.Table(dynamodb_config['tableName'])
  return table.put_item(Item=item)

File: serverless/serverless-node/src/services/dynamodb/currencies.js
const ddb = require("../../utils/db/init-ddb.js");
const { dynamodbConfig: { bpiTableName } } = require("../../config");

const putCurrency = async ({ priceCurrency, currencyCode }) => {
  const extParams = {
    TableName: bpiTableName,
    Item: {
      id: currencyCode,
      ...priceCurrency.data
    }
  };
  return ddb.put(extParams).promise();
};

const getCurrencies = async (
  params = { TableName: bpiTableName },
  allData = []
) => {
  const data = await ddb.scan(params).promise();
  return (data.LastEvaluatedKey)
    ? getCurrencies(
      { ...params, ExclusiveStartKey: data.LastEvaluatedKey },
      [...allData, ...data.Items]
    )
    : [...allData, ...data.Items];
};

module.exports = {
  putCurrency,
  getCurrencies
};
